src/tools/compiletest/src/header.rs

```diff
@@ -574,12 +574,14 @@ impl Config {
         if line.starts_with(prefix) && line.as_bytes().get(prefix.len()) == Some(&b'-') {
             let name = line[prefix.len()+1 ..].split(&[':', ' '][..]).next().unwrap();
 
+            let triple: Vec<&str> = self.target.split('-').collect();
+
             name == "test" ||
-                util::matches_os(&self.target, name) ||             // target
-                name == util::get_arch(&self.target) ||             // architecture
-                name == util::get_pointer_width(&self.target) ||    // pointer width
-                name == self.stage_id.split('-').next().unwrap() || // stage
-                Some(name) == util::get_env(&self.target) ||        // env
+                util::matches_os(&self.target, triple.get(2), name) || // target
```

Comment:
I was always worried about "hard-coding" the meaning of triple components like this since they are somewhat flexible. Since it turns out we even have triples in-tree for which this breaks down, let's not do that...